beacon_node/lighthouse_network/src/types/globals.rs

```diff
@@ -183,6 +184,14 @@ impl<E: EthSpec> NetworkGlobals<E> {
             .collect::<Vec<_>>()
     }
 
+    /// Returns the TopicConfig to compute the set of Gossip topics for a given fork
+    pub fn topic_config(&self) -> TopicConfig {
```

Comment:
we can avoid double iteration by not collection bellow into a `Vec` and just using a reference to `Hashset`. 
`fork_core_topics` can still iterate the `sampling_subnets` normally, see https://github.com/jxs/lighthouse/commit/09c5f6f082514ca3f66503bf5b251a557cd931fc if you want to cherry pick the commit.
notice the `as_topic_config` as [code convention](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv)

Reply:
Thanks! cherry-picked